metrics-core/src/main/java/com/codahale/metrics/LockFreeExponentiallyDecayingReservoir.java

```diff
@@ -64,21 +65,18 @@
         private void update(long value, long timestampNanos) {
             double itemWeight = weight(timestampNanos - startTick);
             double priority = itemWeight / ThreadLocalRandom.current().nextDouble();
+            long currentCount = count.get();
```

Comment:
Need to always increase count here I think

Reply:
```suggestion
            long currentCount = count.incrementAndGet();
```

Reply:
I leveraged the delayed update to avoid the isEmpty check on the map, because increments only occur after elements are successfully added.

You're correct that in the case new sample priority is lower than the existing minimum we no longer increment the value, however the only place the `count` value is used outside of update is `Reservoir.size()`, where the implementation uses the minimum of count and size `return (int) Math.min(size, state.count.get());`, so it isn't impacted.

When I add an else statement to increment the count in all cases, the time jumps from 70ns/op back to 700ns/op!

example I used for measurement:
```diff
private void update(long value, long timestampNanos) {
    double itemWeight = weight(timestampNanos - startTick);
    double priority = itemWeight / ThreadLocalRandom.current().nextDouble();
    long currentCount = count.get();
    if (currentCount < size || values.firstKey() < priority) {
        addSample(priority, value, itemWeight);
+   } else {
+       count.incrementAndGet();
    }
}
```

Reply:
Makes sense, the count is only needed to determine when it's time to start removing from the map, so I think this change looks good

Reply:
Thanks :-)
I'll merge this into the upstream PR branch.